Assets/Scripts/Ability.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class Ability
+{
+    private GameObject player;
+    private Animator animator;
+
+    public string name;
+    public float coolDown;
+    public float damageToEnemy;
+    private string triggerName;
+
+    public float timeOfLastUse;
+
+    public bool unlocked =  true;
+    
+
+    public Ability(string name, GameObject player, float damageToEnemy, float coolDown, string triggerName)
+    {
+        this.name = name;
+        this.player = player;
+        this.damageToEnemy = damageToEnemy;
+        this.coolDown = coolDown;
+        animator = player.GetComponent<Animator>();
+    }
+
+    public bool useAbility(Destructible target)
+    {
+        //check to see if the effect has cooled down
+        if (Time.realtimeSinceStartupAsDouble - timeOfLastUse < coolDown) return false;
```

Comment:
might be a good idea to also add a isUnlocked check to prevent the ability being used before its unlocked

Reply:
I just added an if statement so that if the ability is unlocked tryUseAbility will return false automatically. I will also add the getter method for other possible functionality purposes.